src/lib/search-and-replace.js

```diff
@@ -86,24 +114,30 @@ export const searchAndReplace = async (
 	const replacements = flatten( replacementsArr );
 	debug( 'Pairs: ', pairs, 'Replacements: ', replacements );
 
-	// Get a path for a tmp copy of the input file
-	const { inputFile, outputFile } = inPlace
-		? await inPlaceReplacement( filename )
-		: outputFileReplacement( filename );
+	if ( inPlace ) {
+		await confirm(
+			[],
+			'Are you sure you want to run search and replace on your input file? This operation is not reversible.'
+		);
+	}
 
-	console.log( chalk.blueBright( 'Input File Path: ' ), inputFile );
-	console.log( chalk.blueBright( 'Output File Path: ' ), outputFile );
+	const tmpDir = fs.mkdtempSync( path.join( os.tmpdir(), 'vip-search-replace-' ) );
+	debug( `Created a directory to hold temporary files: ${ tmpDir }` );
 
-	const readStream = fs.createReadStream( inputFile, { encoding: 'utf8' } );
-	const writeStream = fs.createWriteStream( outputFile, { encoding: 'utf8' } );
+	const { readStream, writeStream } = getReadAndWriteStreams( {
+		filename,
+		inPlace,
+		output,
+		tmpDir,
+	} );
 
 	const replacedStream = await replace( readStream, replacements, binary );
 	const result = await new Promise( ( resolve, reject ) => {
 		replacedStream
 			.pipe( writeStream )
 			.on( 'finish', () => {
 				console.log( chalk.green( 'Search and Replace Complete!' ) );
-				resolve( outputFile );
+				resolve( output );
```

Comment:
It'd be good etiquette to delete our temp files when we're finished with them...
We can include a `cleanUp` callback in this `resolve` statement so the caller can run it when its finished with the data.